### PR TITLE
[EFR32] Adds fix for IPv6 SLAAC DAD process loopback suppression

### PR DESCRIPTION
**Pull request by rosahay-silabs**
Adds fix for IPv6 SLAAC duplicate address detection process. 
The device was not able to reject traffic directed to a multicast group, where the traffic originated from its own interface. 
This fix checks for the *source* and *destination* MAC address to determine loopback multicast traffic.

### CODE REVIEW

File: src/platform/EFR32/wifi/ethernetif.cpp [hunk 1]
```diff
@@ -125,6 +125,16 @@ static void low_level_input(struct netif * netif, uint8_t * b, uint16_t len)
     { /* 60 : LWIP frame alignment */
         len = 60;
     }
+
+    /* Drop packets originated from the same interface and is a multicast destination */
+    if ((netif->hwaddr[0] == b[6] && netif->hwaddr[1] == b[7] && netif->hwaddr[2] == b[8] && netif->hwaddr[3] == b[9] &&
```
Comment (andy31415) [inline src/platform/EFR32/wifi/ethernetif.cpp:130]: nit: could this be written somehow easier to parse for correctness? I do not fully know what b and hwaddr is, but maybe something like this:

```cpp
constexpr size_t kHwAddrLen = 6;

const void *origin_mac = b + kHwAddrLen; // is this true? I named this out of thin air
const void *dest_mac = b; 

if ((memcmp(netif->hwaddr, origin_mac, kHwAddrLen) == 0) 
    && (memcmp(netif->dest_mac, kHwAddrLen) != 0)) {
  return;
}
```

use some variable names that make it clear what b[0] and b[6] and such are.

File: src/platform/EFR32/wifi/ethernetif.cpp [hunk 2]
```diff
@@ -125,6 +125,16 @@ static void low_level_input(struct netif * netif, uint8_t * b, uint16_t len)
     { /* 60 : LWIP frame alignment */
         len = 60;
     }
+
+    /* Drop packets originated from the same interface and is a multicast destination */
```
[the remark this answers is not in the document]
Comment (rosahay-silabs) [reply src/platform/EFR32/wifi/ethernetif.cpp:129]: I have tested this using the following steps
https://github.com/project-chip/connectedhomeip/issues/20979#issuecomment-1190351414
and it passed successfully.

File: src/platform/EFR32/wifi/ethernetif.cpp (hunk 1 above)
Comment (rosahay-silabs) [reply src/platform/EFR32/wifi/ethernetif.cpp]: Incorporated the comment.

File: src/platform/EFR32/wifi/ethernetif.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/platform/EFR32/wifi/ethernetif.cpp:129]: That doesn't answer my question.  My question is what happens if a node has a binding set up from one of its client clusters to a group that includes another endpoint on the same node.  In that case we would in fact want to deliver the group message to the node itself, right?

[the remark this answers is not in the document]
Comment (rosahay-silabs) [reply src/platform/EFR32/wifi/ethernetif.cpp]: I thought of that point but did not think of the said use-case when I made this change. Incorporating this, I am making a change only to drop packets till we get a preferred IPV6 address.

Sample of the packets dropped during the DAD process:
```
00> <efr32 > low_level_input: DROP , [ff:ff:ff:ff:ff:ff]<-[xx:xx:xx:3d:3a:e0] type=35a5
00> <efr32 > low_level_input: DROP , [33:33:ff:3d:3a:e0]<-[xx:xx:xx:3d:3a:e0] type=86dd
00> <efr32 > low_level_input: DROP , [33:33:ff:3d:3a:e0]<-[xx:xx:xx:3d:3a:e0] type=86dd
00> <efr32 > low_level_input: DROP , [33:33:00:00:00:02]<-[xx:xx:xx:3d:3a:e0] type=86dd
00> <efr32 > low_level_input: DROP , [33:33:ff:d7:5c:0d]<-[xx:xx:xx:3d:3a:e0] type=86dd
00> <efr32 > low_level_input: DROP , [33:33:ff:3d:3a:e0]<-[xx:xx:xx:3d:3a:e0] type=86dd
```

[the remark this answers is not in the document]
Comment (rosahay-silabs) [reply src/platform/EFR32/wifi/ethernetif.cpp]: @bzbarsky-apple , does this look good?

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: src/platform/EFR32/wifi/ethernetif.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/platform/EFR32/wifi/ethernetif.cpp]: It addresses my question; I don't know past that.

**Pull request merged**